src/fetcher/use-comment-data.ts

```diff
@@ -0,0 +1,81 @@
+import { gql, GraphQLClient } from 'graphql-request'
+import useSWR from 'swr'
+
+import { GetCommentsQuery } from './graphql-types/operations'
+import { endpoint } from '@/api/endpoint'
+import { hasOwnPropertyTs } from '@/helper/has-own-property-ts'
+
+const query = gql`
+  query getComments($id: Int!) {
+    uuid(id: $id) {
+      ... on ThreadAware {
+        ...getCommentsThreads
+      }
+    }
+  }
+  fragment getCommentsThreads on ThreadAware {
+    threads(trashed: false) {
+      nodes {
+        id
+        archived
+        comments {
+          nodes {
+            id
+            trashed
+            content
+            archived
+            createdAt
+            author {
+              username
+              alias
+              id
+              isActiveAuthor
+              isActiveDonor
+              isActiveReviewer
+            }
+          }
+        }
+      }
+    }
+  }
+`
+
+export type GetCommentsNode = Extract<
+  GetCommentsQuery['uuid'],
+  { threads: any }
```

Comment:
@elbotho using `Extract` in this way enables checking for existence of a certain property